transforms/code/license_check/python/src/license_check_transform.py

```diff
@@ -0,0 +1,132 @@
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+
+
+import json
+from argparse import ArgumentParser, Namespace
+
+import pyarrow as pa
+from data_processing.data_access import DataAccess, DataAccessFactory
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import TransformUtils, get_logger
+from transformer import AllowLicenseStatusTransformer, DenyLicenseStatusTransformer
+
+
+logger = get_logger(__name__)
+
+LICENSE_CHECK_PARAMS = "license_check_params"
+
+
+def _get_supported_licenses(license_file: str, data_access: DataAccess) -> list[str]:
+    logger.info(f"Getting supported licenses from file {license_file}")
+    licenses_list = None
+    try:
+        licenses_list_json, _ = data_access.get_file(license_file)
+        licenses_list = json.loads(licenses_list_json.decode("utf-8"))
+        logger.info(f"Read a list of {len(licenses_list)} licenses.")
+    except Exception as e:
+        logger.error(f"Failed to read file: {license_file} due to {e}")
+    return licenses_list
+
+
+class LicenseCheckTransform(AbstractTableTransform):
+    """It can be used to select the rows/records of data with licenses
+    matching those in the approved/deny list.
+    """
+
+    def __init__(self, config: dict):
+        super().__init__(config)
+
+        self.license_check = config.get(LICENSE_CHECK_PARAMS)
+        if not self.license_check:
+            raise ValueError(f"Invalid Argument: cannot create LicenseCheckTransform object.")
+        logger.debug(f"LICENSE_CHECK_PARAMS: {self.license_check}")
+        self.license_column = self.license_check["license_column_name"]
+        allow_no_license = self.license_check["allow_no_license"]
```

Comment:
In general, all/most error checking of configuration should be done in init().  You should not rely on the argument parser for this since, programmatically it is possible to create the instance without our runtimes and cli parser being used.  This helps the model that says transforms, in python only cases anyway, should be able to be independent of how the configuration is created.

So this means checking for missing/bad config values in init().

Reply:
done